spec/lib/rbac/filterer_spec.rb

```diff
@@ -1286,56 +1287,143 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
       end
 
-      context 'with EvmRole-tenant_administrator' do
-        let(:rbac_tenant) do
-          FactoryBot.create(:miq_product_feature, :identifier => MiqProductFeature::TENANT_ADMIN_FEATURE)
+      context "where the user is in a group that is a super_administrator group" do
+        before do
+          admin_user # ensure we create another user as an admin user
+          other_user # ensure we create another user in another regular group
+
+          user.miq_groups << admin_group
+          user.miq_groups << other_group
         end
 
-        let(:tenant_administrator_user_role) do
-          FactoryBot.create(:miq_user_role, :name => MiqUserRole::DEFAULT_TENANT_ROLE_NAME, :miq_product_features => [rbac_tenant])
+        context "and the current_group is not the super_administrator group" do
+          before { user.current_group = group }
+
+          it "returns users from all of the current user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(User, [user, other_user])
+          end
+
+          it "returns user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
 
-        let!(:super_administrator_user_role) do
-          FactoryBot.create(:miq_user_role, :role => "super_administrator")
+        context "and the current_group is the super_administrator group" do
+          before { user.current_group = admin_group }
+
+          it "returns all users" do
+            get_rbac_results_for_and_expect_objects(User, User.all.to_a)
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, MiqGroup.all.to_a)
+          end
         end
+      end
+
+      context "with a self_service user" do
+        let(:user)  { FactoryBot.create(:user, :role => "user_self_service") }
+        let(:group) { user.miq_groups.detect { |g| g.description == "EvmGroup-user_self_service" } }
 
-        let(:group) do
-          FactoryBot.create(:miq_group, :tenant => default_tenant, :miq_user_role => tenant_administrator_user_role)
+        before do
+          other_user # ensure we create another user in another group
         end
 
-        let!(:user_role) do
-          FactoryBot.create(:miq_user_role, :role => "user")
+        it "returns only the current user" do
+          get_rbac_results_for_and_expect_objects(User, [user])
         end
 
-        let!(:other_group) do
-          FactoryBot.create(:miq_group, :tenant => default_tenant, :miq_user_role => user_role)
+        it "returns only the current group" do
+          get_rbac_results_for_and_expect_objects(MiqGroup, [group])
         end
 
+        context "that is in multiple groups" do
```

Comment:
do we need separate tests for a single group vs multiple groups?

Reply:
Probably not, but I was struggling here.  It felt important to me to have a bunch of tests dedicated to multiple groups, so that future changes could take them into account specifically.